java/spark/src/main/java/com/lancedb/lance/spark/internal/LanceFragmentScanner.java

```diff
@@ -54,7 +54,11 @@ public static LanceFragmentScanner create(
       LanceConfig config = inputPartition.getConfig();
       ReadOptions options = SparkOptions.genReadOptionFromConfig(config);
       dataset = Dataset.open(allocator, config.getDatasetUri(), options);
-      fragment = dataset.getFragments().get(fragmentId);
+      fragment =
+          dataset.getFragments().stream()
```

Comment:
This is a O(n) operation? is it sensitive to the performance here?

Reply:
If we want to use an O(1) operation to get the fragment, it has to build the `dataset.getFragments()` as a hash table and store it in the `LanceInputPartition`.

The `LanceInputPartition` will be serialized in spark and it will cause a lot of memory for a big lance dataset. So I think maybe the O(n) filter is a suitable way here.